### PR TITLE
Add new move and focus modes.

### PR DESCRIPTION
**Pull request by hsanson**
- Add move mode that allows moving buffers between windows.
 - Add focus mode that allows moving the cursor between windows.
 - Add additional command (default e) to cycle between modes.

TODO:

 - Add default key shortcuts to start move and focus modes directly.

### CODE REVIEW

**Comment by simeji** [comment]
@hsanson 
I am so sorry for the late response.
Thank you for your PR.
Your PR is very cool, `move mode` and `focus mode` are very useful.

I think almost OK,  but I want to make it easy to change the mode.
(Now, Resize => Move => Focus => Resize ....)

What do think about below keymappings?

Resize mode(Default)
* Enter: OK , 
* w : Move Mode 
* f : Focus Mode 
* q : Cancel

Move mode
* Enter: OK
* r : Resize Mode 
* f : Focus Mode 
* q : Cancel

Focus Mode
* Enter and  r : Resize Mode 
* w: Move Mode 
* q : Cancel

**Comment by hsanson** [comment]
@simeji  added the additional key shortcuts.

File: plugin/winresizer.vim [hunk 1]
```diff
@@ -85,6 +92,7 @@ let g:winresizer_keycode_right = get(g:, 'winresizer_keycode_right', s:default_k
 let g:winresizer_keycode_finish = get(g:, 'winresizer_keycode_finish', s:default_keycode['finish'])
 let g:winresizer_keycode_cancel = get(g:, 'winresizer_keycode_cancel', s:default_keycode['cancel'])
 let g:winresizer_keycode_escape = get(g:, 'winresizer_keycode_escape', s:default_keycode['escape'])
+let g:winresizer_keycode_escape = get(g:, 'winresizer_keycode_enter', s:default_keycode['enter'])
```
Comment (simeji) [inline plugin/winresizer.vim]: Please change `let g:winresizer_keycode_escape` to  `let g:winresizer_keycode_enter`.

File: plugin/winresizer.vim [hunk 2]
```diff
@@ -74,71 +103,137 @@ let s:codeList = {
...
 
 endfun
 
 fun! s:startReize(commands)
 
+  if g:winresizer_enable == 0
+    return
+  endif
+
+  let s:commands = a:commands
```
Comment (simeji) [inline plugin/winresizer.vim]: Why did you choice `s:commands` ?
I think `let commands` (It's equal `let l:commands`) is better.

**Comment by simeji** [comment]
I tested by MacVim(GUI Vim).
Then I found a bug.

In GUI Vim,  This plugin map keys below.
nnoremap  `g:winresizer_gui_start_key`  `:WinResizerStartResizeGUI<CR>'
nnoremap  `g:winresizer_start_key`  `:WinResizerStartResize<CR>`

If This plugin users use GUI Vim,
`WinResizerStartResizeGUI` is expected to resize MacVim Window and `WinResizerStartResize` is expected to resize the vim windows. ( https://github.com/simeji/winresizer#for-macvimgui-vim )

But, `WinResizerStartResizeGUI` and `WinResizerStartResize` works in same way, now.
I guess `WinResizerStartResizeGUI` should not have modes.

File: plugin/winresizer.vim (hunk 1 above)
Comment (hsanson) [inline plugin/winresizer.vim]: Fixed.

File: plugin/winresizer.vim (hunk 2 above)
Comment (hsanson) [inline plugin/winresizer.vim]: You are correct, fixed.

**Comment by hsanson** [comment]
I use terminal nvim so I did not understand GUI resize mode.  I reverted back so C-A is GUI resize mode.

**Comment by simeji** [comment]
@hsanson 
Thanks for your great works!

**Pull request merged**